src/main/java/sirius/biz/analytics/events/EventRecorder.java

```diff
@@ -221,6 +223,38 @@ public <E extends Event> int countEventsInRange(Class<E> eventType,
         });
     }
 
+    /**
+     * Counts the number of distinct values in the given mapping of events which have occurred based on the given <tt>queryTuner</tt>.
+     * <p>
+     * This automatically marks the query as long-running.
+     *
+     * @param eventType  the type of events to query
+     * @param mapping    the mapping to count distinct values for
+     * @param queryTuner the actual filter to apply. Note that {@link Event#EVENT_DATE} should be filtered, as otherwise
+     *                   the performance will be catastrophic.
+     * @param <E>        the generic types of the entities to query
+     * @return the number of events matching the given filter. Note that we return an <tt>int</tt> here to better match
+     * the API of {@link sirius.kernel.health.metrics.Metrics}.
+     * @throws SQLException in case of a database error
+     */
+    public <E extends Event> int countDistinctValuesInEvents(Class<E> eventType,
+                                                             Mapping mapping,
+                                                             @Nullable Consumer<SmartQuery<E>> queryTuner)
+            throws SQLException {
+        SmartQuery<E> query = oma.select(eventType)
+                                 .aggregationField(Strings.apply("COUNT(DISTINCT %s) as distinctCount",
+                                                                 mapping.getName()));
```

Comment:
Can't we use prepared statements instead of insecure `Strings#apply`?

Reply:
In local tests we got some errors trying to bind arguments in different portions of an SQL statement. I works 100% with arguments, but not always everywhere. Surely a shot could be done here, but since the source is a Mapping object, it's practically impossible to inject a bad statement here.